Packs/QueryAI/Integrations/QueryAI/QueryAI.py

```diff
@@ -0,0 +1,215 @@
+        :rtype: ``Dict[str, Any]``
+        """
+
+        if workflow_params:
+            workflow_params = safe_load_json(workflow_params)
+
+        if connection_params:
+            connection_params = safe_load_json(connection_params)
+        else:
+            connection_params = self.connection_params
+
+        if not alias:
+            alias = self.alias
+
+        return self.queryai_http_request(
+            method='POST',
+            url_suffix='/query',
+            json_data={
+                'query': query,
+                'connection_params': connection_params,
+                'workflow_params': workflow_params,
+                'time_text': time_text,
+                'alias': alias
+            }
+        )
+
+
+def generate_drilldown_url(query, alias, time_text=None, workflow_params=None):
+    base_drilldown = 'https://ai.query.ai/login;'
+    questions_url_param = f'questions={urllib.parse.quote(query)};'
+    alias_url_param = f'alias={urllib.parse.quote(alias)};'
+    drilldown_url = base_drilldown + questions_url_param + alias_url_param
+
+    if time_text:
+        query_duration_url_param = f'queryDuration={urllib.parse.quote(time_text)};'
+        drilldown_url += query_duration_url_param
+    if workflow_params:
+        workflow_params_url_param = f'params={urllib.parse.quote(workflow_params)};'
+        drilldown_url += workflow_params_url_param
+
+    return drilldown_url
+
+
+def queryai_run_query_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """queryai-run-query command: Returns response for the query being run on QueryAI
+
+    :type client: ``Client``
+    :param client: QueryAI client to use
+
+    :type args: ``Dict[str, Any]``
+    :param args:
+        all command arguments, usually passed from ``demisto.args()``.
+        ``args['query']`` query to run
+
+    :return:
+        A ``CommandResults`` object that is then passed to ``return_results``,
+        that contains an alert
+
+    :rtype: ``CommandResults``
+    """
+
+    query = args.get('query', '')
+    connection_params = args.get('connection_params', client.connection_params)
+    workflow_params = args.get('workflow_params', None)
+    time_text = args.get('time_text', None)
+    alias = args.get('alias', client.alias)
+
+    if not query:
+        raise ValueError('Missing argument: "query"')
+
+    result = client.run_query(query=query, connection_params=connection_params, workflow_params=workflow_params,
+                              time_text=time_text, alias=alias)
+
+    drilldown_url = f'### Click here to [see details]({generate_drilldown_url(query, alias, time_text, workflow_params)})'
+    readable_output = tableToMarkdown(f'Query.AI Result for the query: {query}', result['data']) if result['data'] else ''
+    readable_output += drilldown_url
+    reply = {'result': result['data'] if result['data'] else result['reply'], 'markdown_string': readable_output}
```

Comment:
why are we adding the readable output to reply? which will be added to the context

Reply:
This is because for our use case we want to show that readable output in a field as well when called from a playbook, as it may contain some drill-down links to our product and the only way to set field values from a playbook is from the context data. Please see the attached screenshot in the description for reference.